File: mobile/lib/features/context/domain/entities/context.dart
import 'package:equatable/equatable.dart';

/// Domain entity for context
class Context extends Equatable {
  final int code;
  final String label;
  final String module;

  const Context({
    required this.code,
    required this.label,
    required this.module,
  });

  @override
  List<Object?> get props => [code, label, module];
}

File: src/domain/silence/context.rs
use sea_query::Iden;
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use utoipa::ToSchema;

/// Silence context entity from database
#[derive(Debug, Clone, Serialize, Deserialize, FromRow, ToSchema)]
pub struct SilenceContext {
    pub code: i16,
    pub label: String,
}

/// Table identifier for silence_contexts table (sea-query)
#[derive(Iden)]
pub enum SilenceContexts {
    Table,
    Code,
    Label,
}
